src/handlers/web3.ts

```diff
@@ -207,8 +211,12 @@ export const getFlashbotsProvider = async () => {
   );
 };
 
-export const getCachedProviderForNetwork = (network: Network = Network.mainnet) => {
-  return networkProviders[network]!;
+export const getCachedProviderForNetwork = (network: Network = Network.mainnet): StaticJsonRpcProvider | undefined => {
+  const cachedProvider = networkProviders.get(network);
+  if (cachedProvider) {
+    return cachedProvider;
+  }
+  return undefined;
 };
```

Comment:
ain't this the same? 
```suggestion
export const getCachedProviderForNetwork = (network: Network = Network.mainnet): StaticJsonRpcProvider | undefined => {
  return networkProviders.get(network);
};
```

Reply:
Yes good call